frontend/src/hooks/use-debounce/use-debounce.hook.ts

```diff
@@ -0,0 +1,14 @@
+type callbackType = () => void;
+const DEFAULT_DELAY_TIME = 500;
+
+const useDebounce = (callback: callbackType, delay = DEFAULT_DELAY_TIME): callbackType => {
+  let timer: NodeJS.Timeout;
+  return () => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      callback();
+    }, delay);
+  };
+};
+
+export { useDebounce };
```

Comment:
```suggestion
const debounce = (fn, delay) => {
  let timeout = null
  return (...args) => {
    clearTimeout(timeout)
    timeout = setTimeout(() => fn.call(null, ...args), delay)
  }
}
export { debounce }
```

move to `shared/helpers/timeout/debounce`